web/javascript/transmission.js

```diff
@@ -1764,5 +1773,56 @@ Transmission.prototype = {
     onStatsDialogClosed: function () {
         this.hideMobileAddressbar();
         this.togglePeriodicStatsRefresh(false);
+    },
+
+    // Magnet: URI web protocol handler
+    onAddMagnetHandlerClicked: function (event) {
+        if ("registerProtocolHandler" in navigator) {
+            navigator.registerProtocolHandler("magnet", getBaseURL() + "?addtorrent=%s", "Transmission Web");
+        }
+        /* The following is not yet implemented in any browser */
+        if ("registerContentHandler" in navigator) {
+            navigator.registerContentHandler("application/x-bittorrent", getBaseURL() + "?addtorrent=%s", "Transmission Web");
+        }
+        $("#add-magnet-handler,#remove-magnet-handler").parent().toggleClass("ui-helper-hidden");
+    },
+
+    onRemoveMagnetHandlerClicked: function (event) {
+        if ("unregisterProtocolHandler" in navigator) {
+            navigator.unregisterProtocolHandler("magnet", getBaseURL() + "?addtorrent=%s");
+        }
+        /* The following is not yet implemented in any browser */
+        if ("unregisterContentHandler" in navigator) {
+            navigator.unregisterContentHandler("application/x-bittorrent", getBaseURL() + "?addtorrent=%s");
+        }        
+        $("#add-magnet-handler,#remove-magnet-handler").parent().toggleClass("ui-helper-hidden");
+    },
+
+    toggleOpenDialogForMagnetHandler: function() {
+        this.setPref(Prefs._OpenDialogForMagnetHandler, this.prefsDialog.shouldOpenDialogForMagnetHandler());
+    },
+
+    initOpenDialogForMagnetHandler: function() {
+        var openDialog = this[Prefs._OpenDialogForMagnetHandler];
+        $("#open-dialog-for-magnet-handler").prop('checked', openDialog);
+    },
+
+    checkForAddTorrentInUrl: function () {
+        var url = getURLParameter("addtorrent");
+
+        if (url && url.lastIndexOf("magnet:") === 0) {
```

Comment:
I think you meant to write `url.lastIndexOf("magnet:", 0)` here to start the search at 0 thus not iterating over the complete string.